lib/client.js

```diff
@@ -74,11 +74,13 @@ async function get(config) {
     // Allow configuration to override user agent header.
     {'User-Agent': `${pkg.name}/${pkg.version}`},
     httpOptions.headers || {},
-    // Do not allow overriding telemetry.
-    {'Auth0-Client': Buffer.from(JSON.stringify(telemetryHeader)).toString('base64')}
+    // Do not allow overriding telemetry, but allow it to be omitted.
+    config.enableTelemetry && {'Auth0-Client': Buffer.from(JSON.stringify(telemetryHeader)).toString('base64')}
   );
 
-  custom.setHttpOptionsDefaults(httpOptions);
+  client[custom.http_options] = function(options) {
```

Comment:
Have changed how the default http options are set to setting them on a per request basis (https://github.com/panva/node-openid-client/blob/master/docs/README.md#customizing-individual-http-requests) to make this more testable.

This prevents an [issue in testing](https://circleci.com/gh/auth0/express-openid-connect/49), but it could also prevent an issue if there was a use case for creating multiple clients:

```
const clientA = await getClient({ httpOptions: { headers: { 'X-Custom-Header': 'bar' } } });
const clientB = await getClient({});

// Before this change, clientB will make requests with the `X-Custom-Header` even though this wasn't specified
```

This came up in the tests because I was testing for the absence of the `Auth0-Client` header when `enableTelemetry` was set to `false`.

This is because `setHttpOptionsDefaults` in `openid-client` mutates the defaults object which is scoped to the module level and can't be cleared https://github.com/panva/node-openid-client/blob/master/lib/helpers/request.js#L13 this makes it impossible to remove defaults when they're set.